pkg/controllers/vdb/saverestorepoint_reconciler.go

```diff
@@ -0,0 +1,225 @@
+type SaveRestorePoint struct {
+	VRec *VerticaDBReconciler
+	Vdb  *vapi.VerticaDB
+	Log  logr.Logger
+	client.Client
+	Dispatcher     vadmin.Dispatcher
+	PFacts         *PodFacts
+	OriginalPFacts *PodFacts
+	InitiatorIP    string // The IP of the pod that we run vclusterOps from
+}
+
+func MakeSaveRestorePointReconciler(r *VerticaDBReconciler, vdb *vapi.VerticaDB, log logr.Logger,
+	pfacts *PodFacts, dispatcher vadmin.Dispatcher, cli client.Client) controllers.ReconcileActor {
+	pfactsForMainCluster := pfacts.Copy(vapi.MainCluster)
+	return &SaveRestorePoint{
+		VRec:           r,
+		Log:            log.WithName("SaveRestorePoint"),
+		Vdb:            vdb,
+		Client:         cli,
+		Dispatcher:     dispatcher,
+		PFacts:         &pfactsForMainCluster,
+		OriginalPFacts: pfacts,
+	}
+}
+
+// Reconcile will create an archive if the status condition indicates true
+// And will save restore point to the created archive if restorePoint.archive value
+// is provided in the CRD spec
+func (s *SaveRestorePoint) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	// Only proceed if the SaveRestorePointsNeeded status condition is set to true.
+	if !s.Vdb.IsStatusConditionTrue(vapi.SaveRestorePointsNeeded) {
+		return ctrl.Result{}, nil
+	}
+	// Check if deployment is using vclusterops
+	if !vmeta.UseVClusterOps(s.Vdb.Annotations) {
+		s.VRec.Event(s.Vdb, corev1.EventTypeWarning, events.InDBSaveRestorePointNotSupported,
+			"SaveRestorePoint is not supported for admintools deployments")
+		err := vdbstatus.UpdateCondition(ctx, s.VRec.Client, s.Vdb,
+			vapi.MakeCondition(vapi.SaveRestorePointsNeeded,
+				metav1.ConditionFalse, "AdmintoolsNotSupported"),
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Ensure vertica version supports in-database SaveRestorePoint vcluster API
+	vinf, err := s.Vdb.MakeVersionInfoCheck()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !vinf.IsEqualOrNewer(vapi.SaveRestorePointNMAOpsMinVersion) {
+		s.VRec.Eventf(s.Vdb, corev1.EventTypeWarning, events.UnsupportedVerticaVersion,
+			"The Vertica version %q doesn't support create restore points. The minimum version supported is %s.",
+			vinf.VdbVer, vapi.SaveRestorePointNMAOpsMinVersion)
+		err = vdbstatus.UpdateCondition(ctx, s.VRec.Client, s.Vdb,
+			vapi.MakeCondition(vapi.SaveRestorePointsNeeded,
+				metav1.ConditionFalse, "IncompatibleDB"),
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	err = s.PFacts.Collect(ctx, s.Vdb)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	hostIP, ok := s.PFacts.FindFirstUpPodIP(false, "")
+	if !ok {
+		s.Log.Info("No running pod for create restore point. Requeuing.")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	// Check if required attribute is set in spec: restorePoint.archive
+	if s.Vdb.Spec.RestorePoint != nil && s.Vdb.Spec.RestorePoint.Archive != "" {
+		// Always tried to create archive
```

Comment:
When the trigger is set, operator doesn't have the information of if the archive is created or not. So it will always try to create it as mentioned in the JIRA. We will handle the "already exist error" in the following runCreateArchiveVclusterAPI method.